### PR TITLE
Add a test-suite with happy-path and sad-path and a comprehensive documentation.

### PR DESCRIPTION
**Pull request by xmontero**
Hey!

I have created this test-suite in order to make "functional testing" on the add-on. They are not "automated tests" but definitively will help very much in the "manual tests".

Those tests will prove that some of the results in #5 are not working properly. I started this test suite specifically to check the new feature pending to be merged. But I have branched from `master` because -in fact- this is independent of the new feature and the test suite can also be used to test the old functionality.

Joshua, please consider accepting the PR into `master` and then, right after, merge `master` back into `raw-copy`.

All the new files are contained inside the `tests` directory, including the `.gitattributes` that will force the sample `.eml` files to be downloaded in CRLF regardless of the general git settings.

There's a comprehensive README that holds all the relevant information about why those test cases and the links to the relevant documentation (RFCs or whatsoever).

The README will tell you how to test, but in short: Make a `_test` folder in your `Local Folders`, then drag-and-drop all the `.eml` files and test one by one that clicking "Copy Message ID".

### CODE REVIEW

File: tests/test-invalid-weirdcasing-spacecrlf-nonlast.eml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+From: John Doe <jdoe@machine.example>
+To: Mary Smith <mary@example.net>
+Subject: Test invalid weirdcasing spacecrlf nonlast
+mssSAGe-iD: < test invalid
```
Comment (garoose) [inline tests/test-invalid-weirdcasing-spacecrlf-nonlast.eml]: Did you intend for "message" to be spelled wrong on this one?

Comment (xmontero) [reply tests/test-invalid-weirdcasing-spacecrlf-nonlast.eml]: Nope, all them should say "message-id". Good catch. Going to commit the change.

Comment (xmontero) [reply tests/test-invalid-weirdcasing-spacecrlf-nonlast.eml]: Committed at 87deeeb. Closing.

**Review by garoose — APPROVED** [review]
lgtm

**Comment by xmontero** [comment]
> 
> 
> lgtm

Igtm?

**Comment by garoose** [comment]
Oops sorry, that stands for "looks good to me"

**Pull request merged**